go/enclave/enclave.go

```diff
@@ -567,19 +570,20 @@ func (e *enclaveImpl) EstimateGas(encryptedParams common.EncryptedParamsEstimate
 	}
 
 	// extract params from byte slice to array of strings
-	var paramString []interface{}
-	err = json.Unmarshal(paramBytes, &paramString)
+	var paramList []interface{}
+	err = json.Unmarshal(paramBytes, &paramList)
 	if err != nil {
 		return nil, fmt.Errorf("unable to decode EstimateGas params - %w", err)
 	}
 
-	// params must always be [callMsg, block number]
-	if len(paramString) != 2 {
-		return nil, fmt.Errorf("invalid number of params, expected 2 got %d", len(paramString))
+	// params are [callMsg, block number (optional)]
+	// todo: handle blocknum, for now we always assume "latest" as with other methods
+	if len(paramList) == 0 {
+		return nil, fmt.Errorf("required at least one param, but received zero")
```

Comment:
Just had a quick look - I think it's fine to merge as it fixes things, but the API for eth_call assumes a callMsg and a block number always. https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_estimategas

Could the issue be that if something sends a blockNumber:nill the paramList len becomes 1 instead of 2 ?

Reply:
Yeah I think that was it, although I think the problem is when it doesn't send a second param rather than sending nil. External tools don't always send block number when they don't have to which causes problems of we're treating exactly one argument as a special case.